src/ngp_algorithms/ABLWallFrictionVelAlg.C

```diff
@@ -0,0 +1,310 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2019 National Renewable Energy Laboratory.                  */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+#include "ngp_algorithms/ABLWallFrictionVelAlg.h"
+
+#include "BuildTemplates.h"
+#include "master_element/MasterElement.h"
+#include "master_element/MasterElementFactory.h"
+#include "ngp_utils/NgpLoopUtils.h"
+#include "ngp_utils/NgpFieldOps.h"
+#include "ScratchViews.h"
+#include "SolutionOptions.h"
+#include "utils/StkHelpers.h"
+#include "wind_energy/MoninObukhov.h"
+
+#include "stk_mesh/base/Field.hpp"
+
+namespace sierra {
+namespace nalu {
+
+namespace {
+
+template <typename PsiFunc>
+KOKKOS_FUNCTION
+double calc_utau(
```

Comment:
@alanw0 @overfelt Can this function be converted to SIMD version? The convergence criteria seems a bit tricky.

Reply:
At first look, I don't think so. I don't know how it would work if there can be a different number of iterations for different entries within a simd type...

Reply:
@sayerhs Shreyas, 
Would not bother with a SIMD version right now.  I think that for SIMD you just iterate until all are converged.  Still faster than iterating each SIMD entry til convergence separately.  I think the big outer if/then/else at the calling point would need to be pushed into the calc_utau function.  The Tflux==0 version might be tricky but maybe that is a rare case.